tests/components/roku/test_media_player.py

```diff
@@ -459,72 +461,182 @@ async def test_services(
         "291097",
         {
             "contentID": "8e06a8b7-d667-4e31-939d-f40a6dd78a88",
-            "MediaType": "movie",
+            "mediaType": "movie",
         },
     )
 
+    await hass.services.async_call(
+        MP_DOMAIN,
+        SERVICE_SELECT_SOURCE,
+        {ATTR_ENTITY_ID: MAIN_ENTITY_ID, ATTR_INPUT_SOURCE: "Netflix"},
+        blocking=True,
+    )
+
+    assert mock_roku.launch.call_count == 3
+    mock_roku.launch.assert_called_with("12")
+
+    await hass.services.async_call(
+        MP_DOMAIN,
+        SERVICE_SELECT_SOURCE,
+        {ATTR_ENTITY_ID: MAIN_ENTITY_ID, ATTR_INPUT_SOURCE: 12},
+        blocking=True,
+    )
+
+    assert mock_roku.launch.call_count == 4
+    mock_roku.launch.assert_called_with("12")
+
+
+async def test_services_play_media(
+    hass: HomeAssistant,
+    init_integration: MockConfigEntry,
+    mock_roku: MagicMock,
+) -> None:
+    """Test the media player services related to playing media."""
     await hass.services.async_call(
         MP_DOMAIN,
         SERVICE_PLAY_MEDIA,
         {
             ATTR_ENTITY_ID: MAIN_ENTITY_ID,
-            ATTR_MEDIA_CONTENT_TYPE: MEDIA_TYPE_URL,
-            ATTR_MEDIA_CONTENT_ID: "https://awesome.tld/media.mp4",
+            ATTR_MEDIA_CONTENT_TYPE: "blah",
+            ATTR_MEDIA_CONTENT_ID: "https://localhost/media.m4a",
             ATTR_MEDIA_EXTRA: {
-                ATTR_NAME: "Sent from HA",
-                ATTR_FORMAT: "mp4",
+                ATTR_NAME: "Test",
             },
         },
         blocking=True,
     )
 
-    assert mock_roku.play_on_roku.call_count == 1
-    mock_roku.play_on_roku.assert_called_with(
-        "https://awesome.tld/media.mp4",
+    assert mock_roku.play_on_roku.call_count == 0
+
+    await hass.services.async_call(
+        MP_DOMAIN,
+        SERVICE_PLAY_MEDIA,
         {
-            "videoName": "Sent from HA",
-            "videoFormat": "mp4",
+            ATTR_ENTITY_ID: MAIN_ENTITY_ID,
+            ATTR_MEDIA_CONTENT_TYPE: MEDIA_TYPE_MUSIC,
+            ATTR_MEDIA_CONTENT_ID: "https://localhost/media.m4a",
+            ATTR_MEDIA_EXTRA: {ATTR_FORMAT: "blah"},
         },
+        blocking=True,
     )
 
+    assert mock_roku.play_on_roku.call_count == 0
+
+
+@pytest.mark.parametrize(
+    "content_type, content_id, resolved_name, resolved_format",
+    [
+        (MEDIA_TYPE_URL, "http://localhost/media.m4a", "media.m4a", "m4a"),
+        (MEDIA_TYPE_MUSIC, "http://localhost/media.m4a", "media.m4a", "m4a"),
+        (MEDIA_TYPE_MUSIC, "http://localhost/media.mka", "media.mka", "mka"),
+        (MEDIA_TYPE_MUSIC, "http://localhost/media.wma", "media.wma", "wma"),
```

Comment:
`wma` is not supported by Roku anymore, removed on firmware v10.5
https://developer.roku.com/en-gb/docs/references/deprecated-apis.md

Reply:
```suggestion
```